sdk/core/core-http/lib/responseBodyNotFoundError.ts

```diff
@@ -15,4 +24,4 @@ export class ResponseBodyNotFoundError extends RestError {
 
     Object.setPrototypeOf(this, ResponseBodyNotFoundError.prototype);
```

Comment:
@bterlson discouraged me from using this approach in [another PR](https://github.com/Azure/azure-sdk-for-js/pull/3826#discussion_r294386874) and recommended setting `this.name` to the name of the derived error class: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/src/client/errors.ts#L87.

Reply:
I had hedged my bets and just added both approaches (.name sounds like it's the right way to go for the future).

I'll remove it from this one and we can come back to it if there's a case we missed.

Reply:
The good news is that I _also_ set .name ;)

This particular module does support es5 (it looks like the identity module doesn't) so I think I'm stuck with leaving this line in.

Reply:
I believe that this was the reason I changed Identity from ES5 to ES6.  We might need to do the same for `core-http`, but @bterlson will need to make a call on whether it's the right thing to do.

Reply:
Closing the loop here after talking with @bterlson. This module needing to support es5 means we have to keep it. Resolving for now.